### PR TITLE
feat: add cookie to create client request

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

1. the RegisterClientRequest atm does not include the cookie label in the request body.
2. the label is not unique, which can lead to issues when removing clients that have been created from the same app.
 
### Solutions

1. generate random uuid as the cookie label when login and new account register
2.  sso login have the label set as null since it can not be cahnged
3. store the generated uuid as part of the user session
4. use the previously generated uuid as the cookie label when registering clients  

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

[SQPIT-764]: https://wearezeta.atlassian.net/browse/SQPIT-764?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SQPIT-764]: https://wearezeta.atlassian.net/browse/SQPIT-764?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepositoryTest.kt [hunk 1]
```diff
@@ -118,35 +118,60 @@ class RegisterAccountRepositoryTest {
...
         given(registerApi).coroutine {
             register(
-                RegisterApi.RegisterParam.PersonalAccount(email, code, name, password)
+                RegisterApi.RegisterParam.PersonalAccount(
+                    email = email,
+                    emailCode = code,
+                    name = name,
+                    password = password,
+                    cookieLabel = cookieLabel
+                )
```
Comment (gongracr) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepositoryTest.kt:141]: I see here so much potential to adapt to `Arrangement` pattern and save us some precious extra lines  😅

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepositoryTest.kt [hunk 2]
```diff
@@ -118,35 +118,60 @@ class RegisterAccountRepositoryTest {
...
 
-        verify(registerApi).coroutine { register(RegisterApi.RegisterParam.PersonalAccount(email, code, name, password)) }
+        verify(registerApi).coroutine {
+            register(
+                RegisterApi.RegisterParam.PersonalAccount(
+                    email = email,
+                    emailCode = code,
+                    name = name,
+                    password = password,
+                    cookieLabel = cookieLabel
```
Comment (gongracr) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepositoryTest.kt:165]: Not sure if I fully understand these tests. Does this mean that the specified `cookieLabel` that we pass upon client registration should be also returned untouched by the backend?

**Review by gongracr — APPROVED** [review]
Looking great. Just a comment/suggestion and a question to help me understand the hard logic underneath this PR, but nothing blocker. GJ 🍄 💪

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepositoryTest.kt (hunk 1 above)
Comment (MohamadJaara) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepositoryTest.kt:141]: i think so too by i will leave it for another PR

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepositoryTest.kt (hunk 2 above)
Comment (MohamadJaara) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepositoryTest.kt:165]: the label is passed from the clients to BE, and we need to make sure that we store the same label that we used to login/register. and the same label is also used for client registration.

this way when a client get deleted it's cookies will also get deleted

**Pull request merged**